lib/cli.js

```diff
@@ -175,7 +175,7 @@ var cli = {
             engine = new CLIEngine(translateOptions(currentOptions));
             debug("Running on " + (text ? "text" : "files"));
 
-            result = text ? engine.executeOnText(text) : engine.executeOnFiles(files);
+            result = text ? engine.executeOnText(text, currentOptions.stdinFilename) : engine.executeOnFiles(files);
             if (currentOptions.quiet) {
```

Comment:
You would like `currentOptions.stdinFilename` to be defaulted to `""`?
The way it is now if `--stdin-filename` isnt passed then `currentOptions.stdinFilename` will be undefined. 
`filename` will then get defaulted to `"<text>"` [here](https://github.com/tikotzky/eslint/blob/add-stdin-filename-option/lib/cli-engine.js#L225)

Would you like me to take a different approach?